server/homebrew.api.js

```diff
@@ -530,11 +531,11 @@ const api = {
 	}
 };
 
-router.post('/api', checkClientVersion, asyncHandler(api.newBrew));
-router.put('/api/:id', checkClientVersion, asyncHandler(api.getBrew('edit', false)), asyncHandler(api.updateBrew));
-router.put('/api/update/:id', checkClientVersion, asyncHandler(api.getBrew('edit', false)), asyncHandler(api.updateBrew));
-router.delete('/api/:id', checkClientVersion, asyncHandler(api.deleteBrew));
-router.get('/api/remove/:id', checkClientVersion, asyncHandler(api.deleteBrew));
-router.get('/api/theme/:renderer/:id', asyncHandler(api.getThemeBundle));
+router.post('/api', checkClientVersion, dbCheck, asyncHandler(api.newBrew));
+router.put('/api/:id', checkClientVersion, dbCheck, asyncHandler(api.getBrew('edit', false)), asyncHandler(api.updateBrew));
+router.put('/api/update/:id', checkClientVersion, dbCheck, asyncHandler(api.getBrew('edit', false)), asyncHandler(api.updateBrew));
+router.delete('/api/:id', checkClientVersion, dbCheck, asyncHandler(api.deleteBrew));
+router.get('/api/remove/:id', checkClientVersion, dbCheck, asyncHandler(api.deleteBrew));
+router.get('/api/theme/:renderer/:id', dbCheck, asyncHandler(api.getThemeBundle));
```

Comment:
I think we can do something like `router.use(dbCheck);` in one place instead of calling it out in each individual route. Or in app.js do `app.use(dbCheck)`.

What do you think?

Reply:
I had `app.use(dbCheck)` initially, then thought that it might be better to apply it individually, so routes that don't touch the DB don't cause an unnecessary check.
That said, those routes are very limited and not something that would be visited regularly: Change Log, FAQ, Migrate, etc. New and Home technically don't use the DB - unless they save.

At the very least, I think adding `router.use(dbCheck)` will apply the check to all of these API routes.

Reply:
I've applied the change to `router.use(dbCheck)` in the latest commit. It appears to work perfectly.